### ISSUE TITLE
MOVE_SCRIPTS_TO_BOTTOM: additional parameters of script tag removed

### ISSUE THREAD

**Issue by Teutron-flo** [primefaces/primefaces#7882]
**Description**
The function primefaces.MOVE_SCRIPTS_TO_BOTTOM, doesn't move addtional parameters of an script import to the bottom.
It works well when a <script> tag has only src / type attributes, but everything else will be deleted during the move process.

For example:

XHTML:
`<script src="https://primefaces.org/primefaces.js"
type="text/javascript" charset="UTF-8"
data-domain-script="OneParameterValue" />`

With primefaces.MOVE_SCRIPTS_TO_BOTTOM=off the output in the browser is:
`<script src="https://primefaces.org/primefaces.js" 
type="text/javascript" charset="UTF-8" data-domain-script="OneParameterValue"></script>`

With primefaces.MOVE_SCRIPTS_TO_BOTTOM=on:
`<script type="text/javascript" src="https://primefaces.org/primefaces.js">`

**Describe the solution you would like**
There are two possible solutions:
1. Skip imports with additional parameters
2. Change the move process, that addtional parameters are considered.

**Comment by tandraschko** [comment]
would you like to prepare a PR?

**Comment by Teutron-flo** [comment]
I have absolutly no clue how GitHub works, its my first time that I work with it.
I'm on it to find out, how I can help you with the pull request.

**Comment by melloware** [comment]
I feel like its best if we skip scripts if they have any special attributes on them?  Because it seems not possible to consolidate them if you had 2 scripts with conflicting attributes. So i vote for just skipping them from moving them.  If so I can work on that PR?

**Comment by tandraschko** [comment]
yep

**Comment by tandraschko** [comment]
i think we must collect them but with attributes and render them again on the right place

**Comment by melloware** [comment]
Render them at the bottom just as full scripts with their attributes right?

**Comment by tandraschko** [comment]
yep

**Comment by tandraschko** [comment]
a contribution would be welcome here

**Comment by melloware** [comment]
Yep I have to collect all attributes too because I just had OmniFaces add `integrity` and `crossorigin` attributes to scripts so I need to collect those and make sure they are moved as well.

**Comment by melloware** [comment]
@tandraschko I chose to only use attributes of include scripts as it makes sense but for inline scripts it doesn't make sense since we collect them all and output them in 1 giant script block.  So I think just doing the include scripts makes sense.  Not sure if you agree?

**Comment by tandraschko** [comment]
the question is: what should we do with inline scripts with attributes? Merge just with just with other inline scripts and remove all attributes - or create a new inline block with those custom attributes?

do we currently ignore other inline scripts, which are not text/js?

**Comment by melloware** [comment]
So my thought was this...

- Do this for actual included scripts only like my PR is currently doing
- Lets leave inline scripts alone the way they are currently working where you gather them all and put them in 1 script block.  
- having separate inline script blocks and tracking any attribute changes feels messy to me.  
- If someone requests it in the future we can always do it but this ticket was specifically referencing included scripts.

So basically lets "Keep It Simple" until someone has a more complex use case for inline scripts.  It's been 4 years of using MOVE_SCRIPTS_TO_BOTTOM and this is the first time even the included script issue came up. 

Thoughts?

**Comment by melloware** [comment]
Sorry i forgot to answer your second part of the question right now you track the inline script by "key" like "text/javascript". So if you had another inline script type it would track by that and merge those so it would create 1 inline script block "per type".

Three is even an existing unit test testing this scenario.  Looks like it was issue: https://github.com/primefaces/primefaces/issues/3854

```java
  /**
     * https://github.com/primefaces/primefaces/issues/3854
     */
    @Test
    public void testMultipleInlineScriptsDifferentTypes() throws IOException {
        writer.startElement("body", null);
        verify(wrappedWriter).startElement("body", null);

        writer.startElement("script", null);
        writer.writeAttribute("type", "text/javascript", null);
        writer.writeText("javascript1", null);
        writer.endElement("script");

        writer.startElement("script", null);
        // default type is text/javascript
        writer.writeText("javascript2", null);
        writer.endElement("script");

        writer.startElement("script", null);
        writer.writeAttribute("type", "x-shader/x-vertex", null);
        writer.writeText("vertex", null);
        writer.endElement("script");

        writer.endElement("body");

        verify(wrappedWriter, times(2)).startElement("script", null);
        verify(wrappedWriter).write(matches("(?s).*javascript1.*javascript2(?!.*vertex.*).*"));
        verify(wrappedWriter).write(matches("(?s)(?!.*javascript.*).*vertex.*"));
        verify(wrappedWriter, times(2)).endElement("script");
    }
```
